kubetools/kubernetes/api.py

```diff
@@ -51,8 +52,9 @@ def check_if_cronjob_batch_v1_compatible(env, batch_api_version):
     required_k8s_version = version.parse("1.21.0")
     supported_beta_version = version.parse("1.25.0")
 
+    version_pattern = re.compile(r"^v?([^-]+).*$")
     nodes_versions = [
-        node.status.node_info.kubelet_version.replace('v', '')
+        version_pattern.sub(r"\1", node.status.node_info.kubelet_version)
         for node in list_node
     ]
     parsed_versions = map(version.parse, nodes_versions)
```

Comment:
Maybe this whole function would work better rewritten using something like https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/CoreApi.md#get_api_versions. i.e. look at the API versions supported by the cluster rather than rely on the version of k8s

Reply:
I don't have older cluster versions to test this on, but here's an attempt: https://github.com/EDITD/kubetools/pull/147/commits/2da736b2377bfe1dfe2214829f12b31e66eeb5b4

Reply:
I've added a fair bit of refactoring and cleanup. I'll try to test that with various k8s cluster versions

Reply:
First split out: Some cleanup and bug fixes I found while testing. https://github.com/EDITD/kubetools/pull/149

Reply:
Finally, also split out the CronJob / APIVersion refactoring into https://github.com/EDITD/kubetools/pull/150